form/_AutoCompleterMixin.js

```diff
@@ -486,6 +486,16 @@ define([
 			}
 			this.inherited(arguments);
 			aspect.after(this, "onSearch", lang.hitch(this, "_openResultList"), true);
+
+			// if the "clear" button is pressed in IE (provided by <input>), then clear the value
+			// must be async since the clearing of the input element happens after the mouseup event 
+			this.on("mouseup", function (e) {
+				setTimeout(lang.hitch(this, function() {
+					if (!e.target.value) {
+						this.set("item", null);
+					}
+				}), 0); 
+			});
```

Comment:
OK, I'm glad that works.   My next question though is whether or not you can leverage `TextBoxMixin#_onInput()`.

There's a mess of code in TextBoxMixin.js for normalizing the input event, although it may all be useless now since IE8 and IE9 are effectively desupported by Microsoft... OTOH dojo and dijit never officially desupported them.